sherpa/astro/tests/test_astro.py

```diff
@@ -407,7 +407,7 @@ def cmp_thread(fres, abs2, mek1, mek2, covarerr):
         assert covarerr[2] == approx(0.00377755, rel=1e-3)
         assert covarerr[3] == approx(0.00370543, rel=1e-3)
         assert covarerr[4] == approx(0.0016608, rel=1e-3)
-        assert fres.statval == approx(0.0496819, rel=1e-4)
+        assert fres.statval == approx(0.0496819, rel=2e-2)
```

Comment:
I would like to understand when the values are different in this PR but (a) the actual fit values are the same, just the statistics is different and (b) I can't reproduce that outside this test. If I run the same thread separately (`fit-spectrum.py` in the sherpa-test-data repro, then I get different numbers. Not very different (for a science application), but different enough that I can't compare them to the numbers in this test. Which is somewhat surprising and means I don't actually understand what this test tests.
```Python
Python 3.11.7 | packaged by conda-forge | (main, Dec 15 2023, 08:41:36) [Clang 16.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from sherpa.astro.ui import *
   ...: 
   ...: load_pha("data1.pi")
   ...: load_arf("arf1.fits")
   ...: load_rmf("rmf1.fits")
   ...: ignore()
   ...: notice(0.5,7)
   ...: 
   ...: set_source(xsphabs.abs2*(xsmekal.mek1+xsmekal.mek2))
   ...: 
   ...: abs2.nh=1.1
   ...: mek1.kt=0.8
   ...: mek1.norm=0.7
   ...: mek2.kt=2.4
   ...: mek2.norm=1.
   ...: 
   ...: mek1.norm.max=1.e10
   ...: mek2.norm.max=1.e10
   ...: mek1.norm.min=1.e-10
   ...: mek2.norm.min=1.e-10
WARNING: systematic errors were not found in file 'data1.pi'
statistical errors were found in file 'data1.pi'
but not used; to use them, re-read with use_errors=True
dataset 1: 0.00146:14.9504 Energy (keV) -> no data
dataset 1: no data -> 0.4964:7.008 Energy (keV)

In [2]: fit()
Dataset               = 1
Method                = levmar
Statistic             = chi2gehrels
Initial fit statistic = 18742.8
Final fit statistic   = 129.134 at function evaluation 25
Data points           = 446
Degrees of freedom    = 441
Probability [Q-value] = 1
Reduced statistic     = 0.292821
Change in statistic   = 18613.7
   abs2.nH        1.13768      +/- 0.00152752  
   mek1.kT        0.855161     +/- 0.00111268  
   mek1.norm      0.682236     +/- 0.00366547  
   mek2.kT        2.34117      +/- 0.00365177  
   mek2.norm      1.04396      +/- 0.00164369 
```

Reply:
So, this is an old test, and it's written in an interesting style. We have long lost any information on what the tests were meant to be showing - are they intended as regression tests or are they there to exercise a particular piece of functionality.

I've found many of the model fits in these tests look poorly constrained, although here at least the two kT values look significantly different. Maybe @anetasie can remember what this test was meant to be showing.

Always beware the "hidden" XSPEC settings - e.g. get_xsabund() and get_xsxect() - when comparing things.

Reply:
I followed the chain. Differences per bin are of order 1e-8, but because the statistic only looks are the difference between data and model (and not the absolute number) it adds up over several bins, leading to a significant change (1.2e-2) in the reduced statistic. The best fit is identical though, so I think this change is safe to accept.